### PR TITLE
Jv rox 18537 switch e2e tests in stackrox to core bpf switch to core bpf

### PR DESCRIPTION
**Pull request by JoukoVirtanen**
## Description

Sets COLLECTION_METHOD to core_bpf for the CI jobs for which that is desired. A list is given below.

**PRs**

gke-qa-e2e-tests
gke-ui-e2e-tests
gke-nongroovy-e2e-tests
gke-sensor-integration-tests
gke-version-compatibility-tests
eks-qa-e2e-tests
osd-aws-qa-e2e-tests
osd-gcp-qa-e2e-tests
aro-qa-e2e-tests
rosa-qa-e2e-tests
aks-qa-e2e-tests
merge-gke-qa-e2e-tests
merge-gke-ui-e2e-tests
merge-gke-nongroovy-e2e-tests
merge-gke-sensor-integration-tests
powervs-qa-e2e-tests
nightly-powervs-qa-e2e-tests
ibmcloudz-qa-e2e-tests
nightly-ibmcloudz-qa-e2e-tests

ocp-4-10-qa-e2e-tests
ocp-4-10-sensor-integration-tests
ocp-4-10-ui-e2e-tests
ocp-4-10-compliance-e2e-tests
ocp-4-10-merge-qa-e2e-tests
ocp-4-10-merge-ui-e2e-tests
ocp-4-10-merge-compliance-e2e-tests

ocp-4-13-qa-e2e-tests
ocp-4-13-sensor-integration-tests
ocp-4-13-ui-e2e-tests
ocp-4-13-fips-qa-e2e-tests
ocp-4-13-pool-qa-e2e-tests
ocp-4-13-compliance-e2e-tests
ocp-4-13-merge-qa-e2e-tests
ocp-4-13-merge-compliance-e2e-tests

**Nightly**

gke-qa-e2e-tests
gke-oldest-qa-e2e-tests
gke-latest-qa-e2e-tests
gke-ui-e2e-tests
gke-oldest-ui-e2e-tests
gke-latest-ui-e2e-tests
gke-nongroovy-e2e-tests
gke-sensor-integration-tests
gke-oldest-nongroovy-e2e-tests
gke-latest-nongroovy-e2e-tests
gke-version-compatibility-tests
eks-qa-e2e-tests
osd-aws-qa-e2e-tests
osd-gcp-qa-e2e-tests
aro-qa-e2e-tests
aks-qa-e2e-tests
rosa-qa-e2e-tests

ocp-4-10-qa-e2e-tests
ocp-4-10-ui-e2e-tests
ocp-4-10-compliance-e2e-tests

ocp-4-10-qa-e2e-tests
ocp-4-10-fips-qa-e2e-tests
ocp-4-10-ui-e2e-tests
ocp-4-10-compliance-e2e-tests

**Release.x.y**

gke-qa-e2e-tests
gke-ui-e2e-tests
gke-nongroovy-e2e-tests
gke-sensor-integration-tests
gke-version-compatibility-tests
gke-race-condition-qa-e2e-tests
eks-qa-e2e-tests
osd-gcp-qa-e2e-tests
aro-qa-e2e-tests
merge-gke-qa-e2e-tests
merge-gke-ui-e2e-tests
merge-gke-nongroovy-e2e-tests
merge-gke-sensor-integration-tests
merge-gke-version-compatibility-tests
merge-eks-qa-e2e-tests
merge-osd-gcp-qa-e2e-tests
merge-aro-qa-e2e-tests
powervs-qa-e2e-tests

ocp-4-10-qa-e2e-tests
ocp-4-10-sensor-integration-tests
ocp-4-10-ui-e2e-tests
ocp-4-10-compliance-e2e-tests
ocp-4-10-merge-qa-e2e-tests
ocp-4-10-merge-ui-e2e-tests
ocp-4-10-merge-compliance-e2e-tests

ocp-4-13-qa-e2e-tests
ocp-4-13-sensor-integration-tests
ocp-4-13-ui-e2e-tests
ocp-4-13-fips-qa-e2e-tests
ocp-4-13-compliance-e2e-tests
ocp-4-13-merge-qa-e2e-tests
ocp-4-13-merge-fips-qa-e2e-tests
ocp-4-13-merge-ui-e2e-tests
ocp-4-13-merge-compliance-e2e-tests

**Removed tests**
The following tests were removed as they were redundant

PRs
ocp-4-13-core-bpf-qa-e2e-tests 
ocp-4-13-merge-core-bpf-qa-e2e-tests.

Nightlies
ocp-4-13-core-bpf-qa-e2e-tests 
ocp-4-13-merge-core-bpf-qa-e2e-tests.

Release.x.y
ocp-4-13-core-bpf-qa-e2e-tests 
ocp-4-13-merge-core-bpf-qa-e2e-tests.

The core_bpf collection method was recently released as a technical preview. In the next release it will be GA. In order to do that we need more confidence that it is working. E2e tests will be switched over to using core_bpf as the collection method. Some e2e tests will be kept as ebpf and the nightly CI runs will use both core_bpf and ebpf.

See the original issue in stackrox/collector

https://github.com/stackrox/collector/issues/1231

See the proposal for this work https://docs.google.com/document/d/1eS6W8yezfG0gf_o99XU7qTZ0iAWiGj54p8Y_uky9nFw/edit

## Related work

PR to set the default collection method to ebpf https://github.com/openshift/release/pull/41702
PR to add ebpf jobs https://github.com/openshift/release/pull/41704

## Testing

master-nightly-ibmcloudz-qa-e2e-tests (Failed. There is no job script for ibmcloudz-qa-e2e-tests)
master-nightly-powervs-qa-e2e-tests (Failed. power is known to be flaky. Might be an infra issue)
master-aks-qa-e2e-tests (Failed. 3/733. DefaultPoliciesTest: Verify that built-in services don't trigger unexpected alerts expand_more Known flake https://issues.redhat.com/browse/ROX-18651 core_bpf)
master-aro-qa-e2e-tests (Passed. core_bpf)
master-eks-qa-e2e-tests (Passed core_bpf)
master-gke-ebpf-race-condition-qa-e2e-tests (Failed. 3/730. SuspiciousLog: Suspicious entries in log file(s) Not found in Jira. ebpf)
master-gke-nongroovy-e2e-tests (Passed core_bpf)
master-gke-qa-e2e-tests (Failed. 3/730. ImageScanningTest: Image metadata from registry test - ecr-config-only Not found in Jira core_bpf)
master-gke-sensor-integration-tests (Passed
master-gke-ui-e2e-tests (Passed core_bpf)
master-gke-version-compatibility-tests (Passed ebpf)
master-ibmcloudz-qa-e2e-tests (Failed. There is no job script for ibmcloudz-qa-e2e-tests)
master-ocp-4-10-compliance-e2e-tests (Passed core_bpf)
master-ocp-4-10-qa-e2e-tests (Passed. core_bpf)
master-ocp-4-10-sensor-integration-tests (Passed)
master-ocp-4-10-ui-e2e-tests (Passed core_bpf)
master-ocp-4-13-compliance-e2e-tests (Passed core_bpf)
master-ocp-4-13-fips-qa-e2e-tests (Failed. 3/89. Fips is flaky. core_bpf)
master-ocp-4-13-pool-qa-e2e-tests (Passed. ebpf)
master-ocp-4-13-qa-e2e-tests (Passed core_bpf)
release-x.y-aro-qa-e2e-tests (Failed. 4/580. aro is flaky. core_bpf)
release-x.y-eks-qa-e2e-tests (Passed. core_bpf)
release-x.y-gke-ebpf-qa-e2e-tests (Passed. ebpf)
release-x.y-gke-ebpf-race-condition-qa-e2e-tests (Passed. ebpf)
release-x.y-gke-nongroovy-e2e-tests (Passed. core_bpf)
release-x.y-gke-qa-e2e-tests (Passed. core_bpf)
release-x.y-gke-sensor-integration-tests (Passed)
release-x.y-gke-qa-e2e-tests (Passed. core_bpf)
release-x.y-gke-ui-e2e-tests (Passed. core_bpf)
release-x.y-gke-version-compatibility-tests (Passed. ebpf)
release-x.y-ocp-4-13-compliance-e2e-tests (Passed. ebpf)
release-x.y-ocp-4-13-fips-qa-e2e-tests (Failed 9/29. Fips is flaky)
release-x.y-ocp-4-13-qa-e2e-tests (Passed. ebpf)
release-x.y-ocp-4-13-sensor-integration-tests (Failed. 5/68)
release-x.y-osd-gcp-qa-e2e-tests (Passed. core_bpf)
release-x.y-powervs-qa-e2e-tests (Passed)

### Summary comparison

Screenshot from 2023-08-02 14-23-41

Screenshot from 2023-08-02 14-25-44

Screenshot from 2023-08-02 14-26-30

Screenshot from 2023-08-02 14-28-20

Screenshot from 2023-08-02 14-28-53

Screenshot from 2023-08-02 14-48-21

Screenshot from 2023-08-02 14-32-44

### CODE REVIEW

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml [hunk 1]
```diff
@@ -439,6 +439,79 @@ postsubmits:
...
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+      timeout: 5h0m0s
+    labels:
+      ci-operator.openshift.io/cloud: aws
+      ci-operator.openshift.io/cloud-cluster-profile: aws
+      ci-operator.openshift.io/variant: ocp-4-10
+      ci.openshift.io/generator: prowgen
+    max_concurrency: 1
```
Comment (BradLugo) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml]:
```suggestion
    max_concurrency: 6
```

**Review by BradLugo — CHANGES_REQUESTED** [review]
Same deal with the other PR; changing a couple `max_concurrency`s and we're good to go

**Comment by BradLugo** [comment]
To capture some context from our Slack discussion in case anyone comes across this PR:

---

**Brad:**
I could imagine myself being confused if if saw something like this:
```yaml
- as: gke-nongroovy-e2e-tests
  optional: true
  steps:
    env:
      COLLECTION_METHOD: core_bpf
    test:
    - ref: stackrox-e2e
- as: gke-operator-e2e-tests
  optional: true
  steps:
    test:
    - ref: stackrox-e2e
- as: ebpf-qa-e2e-tests
  optional: true
  steps:
    cluster_profile: aws
    env:
      TEST_SUITE: ocp-qa-e2e-tests
      COLLECTION_METHOD: ebpf
    test:
    - ref: stackrox-e2e
    workflow: ipi-aws
  timeout: 5h0m0s
```  
Unless there's a reason (which there totally could be that I'm not seeing), I think the jobs that use their names have the TEST_SUITE should all have the COLLECTION_METHOD the same, and anything that uses a different COLLECTION_METHOD should have that in the name

**Jouko**:
I made it so that all jobs that have ebpf in their names use the ebpf collection method. Otherwise they use the core_bpf collection method, or the collection method is not really something that applies to them.

---

Along with another discussion we had about not wanting to change the default `COLLECTION_METHOD` to `core_bpf` due to pre 4.0 release jobs, I agree with Jouko's response and implementation on this. In the future we can explore having `COLLECTION_METHOD: core_bpf` be the default but not at this time.

**Review by BradLugo — APPROVED** [review]
/lgtm

**Review by gavin-stackrox — APPROVED** [review]
/lgtm
/meow

**Pull request merged**